src/plugins/data_source_management/public/components/edit_data_source/components/update_aws_credential_modal/update_aws_credential_modal.tsx

```diff
@@ -0,0 +1,272 @@
+    if (!confirmNewAccessKey) {
+      invalidReason.push('');
+    } else if (confirmNewAccessKey !== newAccessKey) {
+      invalidReason.push(
+        i18n.translate('dataSourcesManagement.editDataSource.accessKeyNoMatch', {
+          defaultMessage: 'Access keys do not match',
+        })
+      );
+    }
+    setIsConfirmNewAccessKeyValid(invalidReason);
+  };
+
+  const validateConfirmNewSecretKey = () => {
+    const invalidReason: string[] = [];
+    if (!confirmNewSecretKey) {
+      invalidReason.push('');
+    } else if (confirmNewSecretKey !== newSecretKey) {
+      invalidReason.push(
+        i18n.translate('dataSourcesManagement.editDataSource.secretKeyNoMatch', {
+          defaultMessage: 'Secret keys do not match',
+        })
+      );
+    }
+    setIsConfirmNewSecretKeyValid(invalidReason);
+  };
+
+  const renderUpdateAwsCredentialModal = () => {
+    return (
+      <EuiModal onClose={closeUpdateAwsCredentialModal}>
+        <EuiModalHeader>
+          <EuiModalHeaderTitle>
+            <h1>
+              {
+                <FormattedMessage
+                  id="dataSourcesManagement.editDataSource.updateStoredAwsCredential"
+                  defaultMessage="Update stored AWS credential"
+                />
+              }
+            </h1>
+          </EuiModalHeaderTitle>
+        </EuiModalHeader>
+
+        <EuiModalBody>
+          <EuiFormRow>
+            <EuiText size="m" style={{ fontWeight: 300 }}>
+              {
+                <FormattedMessage
+                  id="dataSourcesManagement.editDataSource.updateStoredAwsCredentialDescription"
+                  defaultMessage="Update access key and secret key to reflect accurate aws credential to gain access to the endpoint."
+                />
+              }
+            </EuiText>
+          </EuiFormRow>
+          <EuiSpacer size="m" />
+
+          <EuiForm data-test-subj="data-source-update-aws-credential">
+            {/* Region */}
+            <EuiFormRow
+              label={i18n.translate('dataSourcesManagement.editDataSource.region', {
+                defaultMessage: 'Region',
+              })}
+            >
+              <EuiText size="s" data-test-subj="data-source-update-credential-region">
+                {region}
+              </EuiText>
+            </EuiFormRow>
+
+            {/* updated access key */}
+            <EuiFormRow
+              label={i18n.translate('dataSourcesManagement.editDataSource.newAccessKey', {
+                defaultMessage: 'Updated access key',
+              })}
+              isInvalid={!isNewAccessKeyValid}
+            >
+              <EuiFieldPassword
+                name="updatedAccessKey"
+                data-test-subj="updateStoredAwsCredentialUpdatedAccessKeyField"
+                placeholder={i18n.translate(
+                  'dataSourcesManagement.editDataSource.newAccessKeyPlaceHolder',
+                  {
+                    defaultMessage: 'Updated access Key',
```

Comment:
153, 167 One of this should be 'Confirm updated access key' ?  also`k` lower case on Key.